More test refactors

alexytsu opened the pull request:
- Adds a utility function for serializing cbor params
- Fixes a bug in the test_vm that was causing some message invocations to fail
- More of the easily transferred tests are ported to use the new dyn VM trait

anorth commented on test_vm/tests/evm_test.rs:
```diff
@@ -231,34 +239,46 @@ fn test_evm_create() {
+    evm_eth_create_external_test(&v, &v);
+}
+
+// TODO: drop usage of the concrete TestVM
```
Please describe what's blocking doing it now, and link to an issue to resolve that.

anorth commented on test_vm/tests/evm_test.rs:
```diff
@@ -231,34 +239,46 @@ fn test_evm_create() {
-    let account = v.normalize_address(&eth_addr).unwrap();
-    let mut actor = v.get_actor(&account).unwrap();
+
+    // FIXME: refactor this behaviour in terms of the abstract VM trait
```
Don't commit `FIXME`. Is this just the same issue as the comment on the function signature? Can omit the comment here (move it to the signature) if so.

anorth commented on test_vm/tests/evm_test.rs:
```diff
@@ -351,10 +379,11 @@ fn test_evm_staticcall() {
+            // *assert_eq!(
+            //     &v.get_actor(&Address::new_id(create_return.actor_id)).unwrap().code,
+            //     &*EVM_ACTOR_CODE_ID
+            // );
```
This isn't ready to merge if the test is weakened.

anorth commented on test_vm/tests/extend_sectors_test.rs:
```diff
@@ -130,43 +134,46 @@ fn extend<BS: Blockstore>(
-    let store = MemoryBlockstore::new();
-    let v = TestVM::<MemoryBlockstore>::new_with_singletons(&store);
-    let addrs = create_accounts(&v, 3, &TokenAmount::from_whole(10_000));
+// TODO: remove usage of _v_concrete for mutate_state
```
Please link to an issue

anorth commented on test_vm/tests/extend_sectors_test.rs:
```diff
@@ -213,9 +220,11 @@ fn extend_legacy_sector_with_deals_inner(do_extend2: bool) {
     // Manually craft state to match legacy sectors
-    v.mutate_state(&miner_id, |st: &mut MinerState| {
-        let mut sectors = Sectors::load(&store, &st.sectors).unwrap();
+    // FIXME: remove usage of concrete VM from this test
```
Remove FIXME, the comment at the top is sufficient.

anorth reviewed (commented):
The EVM test isn't ready to merge yet, since an assertion has been dropped. Either update the trait to support this (in another PR ahead of this one), or drop the EVM test from this PR, and the rest is good to go.

alexytsu commented:
Thanks, the `get_actor` functionality came in a later commit and I dropped commit that updated the test in a rebase. 

I've also added tracking issues look at removing uses of the concrete TestVM when various mutation methods are needed.
